apps/playwright/tests/admin/auth.spec.ts

```diff
@@ -0,0 +1,164 @@
+import { Page } from "@playwright/test";
+
+import { test, expect } from "~/fixtures";
+import auth from "~/constants/auth";
+import { getAuthCookies, getAuthTokens, AuthCookies } from "~/utils/auth";
+
+function expectAuthCookies(cookies: AuthCookies) {
+  expect(cookies).toEqual(
+    expect.objectContaining({
+      apiSession: expect.objectContaining({ name: "api_session" }),
+      xsrf: expect.objectContaining({ name: "XSRF-TOKEN" }),
+    }),
+  );
+}
+
+const restrictedPaths = [
+  "/en/admin/dashboard",
+  "/en/admin/talent-requests",
+  "/en/admin/users",
+  "/en/admin/settings/classifications",
+  "/en/admin/pools",
+  "/en/admin/settings/departments",
+  "/en/admin/settings/skills",
+  "/en/admin/settings/skills/families",
+];
+
+test.describe("Anonymous user", () => {
+  test("Redirects restricted pages to sign in", async ({ browser }) => {
+    await Promise.all(
+      restrictedPaths.map(async (restrictedPath) => {
+        const context = await browser.newContext();
+        const page = await context.newPage();
+        await page.goto(restrictedPath);
+        await page.waitForURL("**/login-info*");
+        await expect(page.url()).toContain("/en/login-info");
+      }),
+    );
+  });
+
+  test("Redirects app login page to auth login page", async ({ page }) => {
+    await page.goto("/login");
+    await page.waitForURL(`**${auth.SERVER_ROOT}/authorize*`);
+    await expect(page.url()).toContain(auth.SERVER_ROOT + "/authorize");
+  });
+
+  test("Does not have tokens", async ({ appPage }) => {
+    await appPage.gotoHome();
+    const tokens = await getAuthTokens(appPage.page);
+
+    expect(tokens.idToken).toBeNull();
+    expect(tokens.accessToken).toBeNull();
+    expect(tokens.refreshToken).toBeNull();
+  });
+
+  test("Does not have cookies", async ({ appPage }) => {
+    await appPage.gotoHome();
+    const { apiSession, xsrf } = await getAuthCookies(appPage.page);
+
+    expect(apiSession).toBeUndefined();
+    expect(xsrf).toBeUndefined();
+  });
+
+  test("Sets cookies on login redirect", async ({ request, page }) => {
+    let cookies = await getAuthCookies(page);
+
+    expect(cookies.apiSession).toBeUndefined();
+    expect(cookies.xsrf).toBeUndefined();
+
+    await page.goto("/login");
+    cookies = await getAuthCookies(page);
+    expectAuthCookies(cookies);
+  });
+});
+
+test.describe("Authenticated", () => {
+  test("Has tokens", async ({ applicantPage }) => {
+    await applicantPage.gotoHome();
```

Comment:
So if you use the applicantPage fixture, it always comes authenticated? That's cool! Where does it actually happen? `fixtures/ApplicantPage.ts` doesn't look like it does anything...

Reply:
Yeah! I really liked this. We can also do other roles too! It happens in the `global.setup.ts` file before everything runs.

We log the user in then it takes a snapshot of the current app state (cookies, etc.) and stores it in a JSON file under `.auth`. Then when you use this fixture, it loads that into the browser. As mentioned before, it gets stale during testing since this seem to only happen once. I offered some solutions to attempt to keep this file up to date though.

https://github.com/GCTC-NTGC/gc-digital-talent/blob/95c8c0d584e44f6f96dba759e8c6388b6137fd1d/apps/playwright/global.setup.ts#L6-L14